### PR TITLE
Features/refactor

### PR DESCRIPTION
**Pull request by Alzubair98**
all requirements are done 

1-Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
2-Follow SOLID principles.
3-Pay special attention to the "Single-responsiblity Principle".
4-Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -92,7 +92,7 @@ def book_list
 
 def people_list
   puts 'Choose a person from the list by number:-'
-  i = 0
+  j = 0
```
Comment (ShahierNashaat) [inline app.rb]: - [x] Kindly change this line from `j = 0` to `i = 0` to make the app work as it rising errors because of this.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Alzubair98, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Alzubair98) [reply app.rb]: oh 
thats funny 
i did this in the bug challange next week and i forget to change it 
i will change it now thanks 
(💯

**Review by ShahierNashaat — APPROVED** [review]
Hi @Alzubair98, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**